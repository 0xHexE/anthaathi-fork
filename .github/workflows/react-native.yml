name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
