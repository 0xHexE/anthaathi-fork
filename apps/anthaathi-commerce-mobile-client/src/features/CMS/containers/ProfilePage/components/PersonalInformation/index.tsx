import {Text, View} from 'react-native';
import React from 'react';
import {Button, Card} from 'react-native-paper';
import {useIntl} from 'react-intl';
import {ProfilePageComponentType} from '../../../../types/common';

type PersonalInfo = {
  name: string;
  email: string;
  mobile: string;
  alternateMobile: string;
};

export interface PersonalInformationProps {
  title: string;
  handlePress?: () => void;
  personalInfo: PersonalInfo;
}

const PersonalInformation = (props: PersonalInformationProps) => {
  const intl = useIntl();

  return (
    <Card
      style={{
        marginHorizontal: 10,
        borderColor: '#E3E2E7',
        borderWidth: 1,
        borderRadius: 4,
        marginVertical: 5,
      }}
      testID="personalInformation">
      <Card.Content>
        <View
          style={{
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
          }}>
          <Text style={{color: '#364A15', fontSize: 16, fontWeight: '600'}}>
            {props.title}
          </Text>
          <Button
            testID="editPersonalInfo"
            mode="text"
            labelStyle={{color: '#008D3E'}}
            onPress={props.handlePress}>
            Edit
          </Button>
        </View>
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Name'})}
          name={props.personalInfo.name}
        />
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Email'})}
          name={props.personalInfo.email}
        />
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Mobile'})}
          name={props.personalInfo.mobile}
        />
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Alternate Mobile'})}
          name={props.personalInfo.alternateMobile}
        />
      </Card.Content>
    </Card>
  );
};

const InformationRenderer = ({label, name}: {label: string; name: string}) => {
  return (
    <View
      testID="infoTextComponentId"
      style={{
        flexDirection: 'row',
        marginVertical: 5,
      }}>
      <Text style={{color: '#808080', fontSize: 14, fontWeight: '400'}}>
        {label + ':  '}
      </Text>
      <Text style={{color: '#364A15', fontSize: 14, fontWeight: '600'}}>
        {name}
      </Text>
    </View>
  );
};

export default PersonalInformation;

export const PersonalInformationCMSInput = {
  _component: ProfilePageComponentType.PersonalInformation,
  component: PersonalInformation,
};
