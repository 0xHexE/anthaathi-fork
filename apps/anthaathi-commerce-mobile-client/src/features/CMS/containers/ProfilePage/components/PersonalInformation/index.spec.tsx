import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import PersonalInformation from './index';

describe('PersonalInformation', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PersonalInformation
            title={'Personal Information'}
            personalInfo={{
              name: 'User name',
              email: 'user_mail@gmail.com',
              mobile: '+91 0000000000',
              alternateMobile: '+91 0000000000',
            }}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('personalInformation')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PersonalInformation
            title={'Personal Information'}
            personalInfo={{
              name: 'User name',
              email: 'user_mail@gmail.com',
              mobile: '+91 0000000000',
              alternateMobile: '+91 0000000000',
            }}
            handlePress={onpress}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('editPersonalInfo')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have PersonalInformation title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PersonalInformation
            title={'Personal Information'}
            personalInfo={{
              name: 'User name',
              email: 'user_mail@gmail.com',
              mobile: '+91 0000000000',
              alternateMobile: '+91 0000000000',
            }}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Personal Information')).toBeTruthy();
  });

  it('should have PersonalInformation Text Component', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PersonalInformation
            title={'Personal Information'}
            personalInfo={{
              name: 'User name',
              email: 'user_mail@gmail.com',
              mobile: '+91 0000000000',
              alternateMobile: '+91 0000000000',
            }}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryAllByTestId('infoTextComponentId').length).toBe(4);
  });
});
