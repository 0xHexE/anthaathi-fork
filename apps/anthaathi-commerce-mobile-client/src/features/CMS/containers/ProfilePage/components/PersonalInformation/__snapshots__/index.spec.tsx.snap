// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInformation should render the item 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "borderColor": "#E3E2E7",
          "borderRadius": 4,
          "borderWidth": 1,
          "elevation": 1,
          "marginHorizontal": 10,
          "marginVertical": 5,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="personalInformation"
      >
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#364A15",
                  "fontSize": 16,
                  "fontWeight": "600",
                }
              }
            >
              Personal Information
            </Text>
            <View
              collapsable={false}
              style={
                Object {
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "transparent",
                      "borderRadius": 20,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "minWidth": 64,
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": true,
                      }
                    }
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 20,
                        },
                      ]
                    }
                    testID="editPersonalInfo"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "rgba(28, 27, 31, 1)",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Object {
                              "writingDirection": "ltr",
                            },
                            Array [
                              Object {
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              false,
                              Object {
                                "marginHorizontal": 12,
                              },
                              undefined,
                              false,
                              Object {
                                "color": "rgba(103, 80, 164, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              Object {
                                "color": "#008D3E",
                              },
                            ],
                          ]
                        }
                      >
                        Edit
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginVertical": 5,
              }
            }
            testID="infoTextComponentId"
          >
            <Text
              style={
                Object {
                  "color": "#808080",
                  "fontSize": 14,
                  "fontWeight": "400",
                }
              }
            >
              Name:  
            </Text>
            <Text
              style={
                Object {
                  "color": "#364A15",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              User name
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginVertical": 5,
              }
            }
            testID="infoTextComponentId"
          >
            <Text
              style={
                Object {
                  "color": "#808080",
                  "fontSize": 14,
                  "fontWeight": "400",
                }
              }
            >
              Email:  
            </Text>
            <Text
              style={
                Object {
                  "color": "#364A15",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              user_mail@gmail.com
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginVertical": 5,
              }
            }
            testID="infoTextComponentId"
          >
            <Text
              style={
                Object {
                  "color": "#808080",
                  "fontSize": 14,
                  "fontWeight": "400",
                }
              }
            >
              Mobile:  
            </Text>
            <Text
              style={
                Object {
                  "color": "#364A15",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              +91 0000000000
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginVertical": 5,
              }
            }
            testID="infoTextComponentId"
          >
            <Text
              style={
                Object {
                  "color": "#808080",
                  "fontSize": 14,
                  "fontWeight": "400",
                }
              }
            >
              Alternate Mobile:  
            </Text>
            <Text
              style={
                Object {
                  "color": "#364A15",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              +91 0000000000
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
